.github/workflows/pkg.yml

```diff
@@ -0,0 +1,331 @@
+# GitHub Actions workflow for building and testing Routinator O/S packages.
+# Uses GitHub Actions caching to avoid rebuilding Rust cargo-deb and
+# Routinator dependencies on every run.
+#
+# Note: at the time of writing the GH cache contents expire after a
+# week if not used so the next build may be much slower as it will
+# have to re-download/build/install lots of Rust crates.
+#
+# Packages are built inside Docker containers as GH Runners have extra libraries
+# and packages installed which can cause package building to succeed but package
+# installation on a real target O/S to fail, due to being built against too
+# recent version of a package such as libssl or glibc.
+#
+# Packages are tested inside LXC/LXD containers because Docker containers don't
+# by default support init managers such as systemd but we want to test systemd
+# service unit installation and activation.
+
+name: Packaging
+on: [push, pull_request]
+
+defaults:
+  run:
+    # see: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+jobs:
+  # Use the cargo-deb Rust crate to build a Debian package for installing
+  # Routinator. See: https://github.com/mmstick/cargo-deb
+  deb-pkg:
+    strategy:
+      matrix:
+        image: [
+          "ubuntu:16.04",
+          "ubuntu:18.04",
+          "ubuntu:20.04",
+          "debian:9",
+          "debian:10",
+          "debian:bullseye", # there is no debian:11 Docker Hub image tag yet
+        ]
+    env:
+      CARGO_DEB_VER: 94ba8f3
+      # A Routinator version of the form 'x.y.z-bis' denotes a dev build that is
+      # newer than the released x.y.z version but is not yet a new release.
+      NEXT_VER_LABEL: bis
+    name: deb-pkg
+    runs-on: ubuntu-latest
+    # Build on the oldest platform we are targeting in order to avoid
+    # https://github.com/rust-lang/rust/issues/57497. Specifying container
+    # causes all of the steps in this job to run inside a Docker container.
+    container: ${{ matrix.image }}
```

Comment:
I can't think of a way to make this expression handle the conversion from debian:11 to debian:bullseye. Without that we either have all numeric and one inconsistent named "bullseye" version due to the not-yet-present debian:11 image tag on Docker Hub, or perhaps we should switch all of the image values to name based instead of numeric based references to make them consistent. However, I at least, find numeric values easier to reason about than having to remember which image name is which version.

Reply:
> perhaps we should switch all of the image values to name based instead of numeric based references to make them consistent

https://hub.docker.com/_/debian

Personally, I rather use the normal release names and include a comment with the release number (or [just a link](https://www.debian.org/releases/) to external documentation). Rather than the numbering scheme. Makes targeting OS releases other than the stable releases way easier.

Reply:
Perhaps it's a personal preference thing, e.g. I run Ubuntu 18.04, I can't tell you what the release name is, I have no idea without looking it up. However, I think in this case I would rather have consistently named jobs and so may be forced to use release names. I'll give it some thought and return to this tomorrow.